otx/algorithms/segmentation/tasks/inference.py

```diff
@@ -78,6 +78,9 @@ def __init__(self, task_environment: TaskEnvironment):
         self.freeze = True
         self.metric = "mDice"
         self._label_dictionary = {}  # type: Dict
+        self.train_type = None
```

Comment:
If `train_type` is used globally and it is never modified except for L206, how about `self.train_type = self._hyperparams.algo_backend.train_type` here instead of L190 and L213?

Reply:
It won't work because self._hyperparams is not declared yet. The reason why I defined train_type here is to avoid pylint attribute-defined-outside-init error.

Reply:
Wouldn't this be possible by declaring it after super().__init__?

Reply:
I don't know whether to necessarily change it.. Could you explain why?

Reply:
I don't know why we need to avoid the attribute-defined-outside-init error. Does train_type affect super().__init__?

Reply:
Actually, self.train_type is re-defined inside in the _init_recipe. If I add the self.train_type code in init, it's code duplication. I rather chose to avoid attribute-defined-outside-init error here by ignoring.